lib/Doctrine/DBAL/Tools/Console/Command/PingCommand.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Doctrine\DBAL\Tools\Console\Command;
+
+use Doctrine\DBAL\Connection;
+use RuntimeException;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Throwable;
+use function is_numeric;
+use function sleep;
+use function sprintf;
+
+class PingCommand extends Command
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected static $defaultName = 'dbal:ping';
+
+    protected function configure()
+    {
+        $this
+            ->setDescription('Check db is available')
+            ->addOption('limit', null, InputOption::VALUE_REQUIRED, 'Max number of pings to try', '1')
+            ->addOption('sleep', null, InputOption::VALUE_REQUIRED, 'Length of time (seconds) to sleep between pings', '1');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output) : int
+    {
+        $limit = $input->getOption('limit');
+        if (! is_numeric($limit)) {
+            throw new RuntimeException("Option 'limit' must contain an integer value");
+        }
+        $sleep = $input->getOption('sleep');
+        if (! is_numeric($sleep)) {
+            throw new RuntimeException("Option 'sleep' must contain an integer value");
+        }
+
+        return $this->waitForPing($this->getHelper('db')->getConnection(), (int) $limit, (int) $sleep, $output);
+    }
+
+    /**
+     * @return int > 0 for error
+     */
+    private function waitForPing(Connection $conn, int $limit, int $sleep, OutputInterface $output) : int
+    {
+        while (true) {
+            $last = $this->ping($conn, $output);
+            if ($last === 0 || --$limit === 0) {
+                return $last;
```

Comment:
This could be replaced with `break` to avoid the duplication of the `return $last` logic and allow for further changes.